notebooks/openvino/phi-2_on_mtl.ipynb

```diff
@@ -0,0 +1,585 @@
+    "* `save_name`: directory where the exported & quantized model will be saved.\n",
+    "* `precision`: the compute data type we will use for inference of the model, can be either `f32` or `f16`. We use FP32 precision due to Phi-2 overflow issues in FP16.\n",
+    "* `quantization_config`: here we set the attributes for the weight only quantization algorithm:\n",
+    "    * `bits`: number of bits to use for quantization, can be either `8` or `4`.\n",
+    "    * `sym`: whether to use symmetric quantization or not, can be either `True` or `False`.\n",
+    "    * `group_size`: number of weights to group together for quantization. We use groups of 128 to ensure no accuracy degradation.\n",
+    "    * `ratio`: the ratio of the model to quantize to #`bits`. The rest will be quantize to the default bits number, `8`.\n",
+    "* `device`:  the device to use for inference, can be either `cpu` or `gpu`.\n",
+    "* `stateful`: Optimize model by setting the KV cache as part of the models state instead of as an input\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "800cd7a3-a21d-4a0a-9d73-2a2d08646f99",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_name = 'microsoft/phi-2'\n",
+    "save_name = './phi-2-woq4'\n",
+    "precision = 'f32'\n",
+    "quantization_config = OVWeightQuantizationConfig(\n",
+    "    bits=4,\n",
+    "    sym=False,\n",
+    "    group_size=128,\n",
+    "    ratio=0.8,\n",
+    ")\n",
+    "device = 'gpu'\n",
+    "stateful = True "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1f398868-93d7-4c2d-9591-9bac8e9b701c",
+   "metadata": {},
+   "source": [
+    "With this configuration we expect the model size to reduce to around to 1.62GB:  $0.8 \\times 2.7{\\times}10^3 \\times \\frac{1}{2}\\text{B} + 0.2 * 2.7{\\times}10^3 \\times 1\\text{B} = 1.62{\\times}10^3\\text{B} = 1.62\\text{GB}$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d994997d-344c-4d6c-ab08-f78ecb7f56ec",
+   "metadata": {},
+   "source": [
+    "## Export & quantize\n",
+    "OpenVINO together with 🤗 Optimum enables you to load, export and quantize a model in a single `from_pretrained` call making the process as simple as possible.\n",
+    "Then, we will save the exported & quantized model locally on our laptop. If the model was already exported and saved before we will load the locally saved model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "03a308c6-27e7-4926-8ac4-4fa0c1ca68d2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load kwargs\n",
+    "load_kwargs = {\n",
+    "    'device': device,\n",
+    "    'ov_config': {\n",
+    "        \"PERFORMANCE_HINT\": \"LATENCY\",\n",
+    "        \"INFERENCE_PRECISION_HINT\": precision,\n",
+    "        \"CACHE_DIR\": os.path.join(save_name, \"model_cache\"),  # OpenVINO will use this directory as cache\n",
+    "    },\n",
+    "    'stateful': stateful,\n",
```

Comment:
Statefulness is enabled by default and there is no reason to disable statefulness, so I would remove this option.

Reply:
We have plans to update this notebook to showcase other uses that require stateful to be off and this kind of left over. We can remove this for now if it is an issue.

Reply:
I would really prefer to remove it in this notebook for now. Curious about the usecase that requires stateful to be off!

Reply:
Removed